tests/test_postproc.py

```diff
@@ -0,0 +1,97 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+"""
+tests.test_postproc- Tests of various functions in newspack, among which rar_renamer()
+"""
+
+from sabnzbd.postproc import *
+from unittest import mock
+import os, shutil, pytest
+from distutils.dir_util import copy_tree
+
+
+class TestPostProc:
+
+    # Tests of rar_renamer() (=deobfuscate) against various input directories
+    def test_rar_renamer(self):
+
+        # Function to deobfuscate one directorty with rar_renamer()
+        def deobfuscate_dir(sourcedir, expected_filename_matches):
+            # sourcedir is the relative path to the directory with obfuscated files
+
+            # enrich to absolute path:
+            sourcedir = os.path.join(os.getcwd(), sourcedir)
+            # We create a workingdir inside the sourcedir, because the filenames are really changed
+            workingdir = os.path.join(sourcedir, "workingdir")
+            #print("workingdir is", workingdir)
+            # if workingdir is still there from previous run, remove it:
+            if os.path.isdir(workingdir):
+                try:
+                    shutil.rmtree(workingdir)
+                except PermissionError:
+                    pytest.fail(
+                        "Could not remove existing workingdir %s for rar_renamer",
+                        workingdir,
+                    )
```

Comment:
You can't use it like this. You are giving `workingdir` as the second argument for `fail` instead of as a print-input. Only `logging.info` etc work like this.
It should be
```python
 pytest.fail("Could not remove existing workingdir %s for rar_renamer" % workingdir)

Reply:
Thanks! Solved.